### PR TITLE
ARROW-1443: [Java] Fixed a small bug on ArrowBuf.setBytes with unsliced ByteBuffers

### PR DESCRIPTION
**Pull request by gortiz**
There is a small bug on ArrowBuf at line 750. It says:
`udle.setBytes(index + offset, buf);`

But it should say:
`udle.setBytes(index + offset, newBuf);`

There is a wraparound: You can call the method with the already sliced buffer.

### CODE REVIEW

**Comment by wesm** [comment]
This seems like a quite serious bug, perhaps this code path is never exercised? Can you add a unit test for this? 

Also, would you mind opening a JIRA for the changelog and add it to the PR title? Thanks!

**Comment by gortiz** [comment]
You can bet it was a serious bug. I spent some time trying to catch it! But it was quite easy to fix.

I have pushed 3 commits: The first one reverts the previous one, the next creates the test and the last reapplies the fix.

You can checkout 41bcacc to verify it was happening ;)

File: java/memory/src/test/java/io/netty/buffer/ArrowBufTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.Arrays;
+import java.util.stream.IntStream;
+import org.apache.arrow.memory.RootAllocator;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class ArrowBufTest {
```
Comment (icexelloss) [inline java/memory/src/test/java/io/netty/buffer/ArrowBufTest.java]: Can you rename this to TestArrowBuf?

**Comment by icexelloss** [comment]
Thanks @gortiz for catching this.

This wasn't covered by any test because the code path is only used by VarBinaryVector and there is no java unit test for VarBinaryVector.

@gortiz can you give me write permission to your branch? I'd like to add a test for VarBinaryVector as well. (If you are willing to add the test it'd be awesome too!)

**Comment by gortiz** [comment]
@icexelloss I'm not sure if there is other ways to give you write permission, but I've added you and @wesm as collaborators on my fork. That should be enough

**Comment by icexelloss** [comment]
@gortiz Thanks. Actually there is test for VarBinaryVector but doesn't cover this code path. I will add test in TestValueVector.java.

**Comment by icexelloss** [comment]
@gortiz I pushed a change that adds more vector tests to your branch. Can you rename ArrowBuf test and add license to the file?

Otherwise LGTM

**Comment by icexelloss** [comment]
@gortiz Can you remove the old ArrowBufTest.java file?

**Comment by icexelloss** [comment]
Otherwise LGTM

**Comment by gortiz** [comment]
Done. My bad!

**Review by wesm — APPROVED** [review]
+1. thanks @icexelloss for reviewing!

**Comment by wesm** [comment]
thank you @gortiz for your contribution!

**Comment by gortiz** [comment]
Happy to help! I hope my employer decides to integrate Arrow in our project, in which case I will be pushing to contribute with more PRs!

**Pull request closed without merging**